src/renderer/hooks/useTestSequencerState.ts

```diff
@@ -212,26 +266,162 @@ export function useTestSequencerState() {
+      }
+    } else {
+      // Run next sequence
+      runNextRunnableSequence(sender);
+    }
+  }
+
+  function handleUpload(aborted: boolean, forceUpload: boolean = false) {
+    if (uploadAfterRun || forceUpload) {
+      if (project === null) {
+        toast.warning("No sequence to upload, please create one.");
+        return;
+      }
+      const upload = async () => {
+        await postSession(
+          uploadInfo.serialNumber,
+          uploadInfo.stationId,
+          uploadInfo.integrity,
+          aborted,
+          "",
+          [...useSequencerStore.getState().cycleRuns],
+        );
+      };
+      toast.promise(upload, {
+        loading: "Uploading result...",
+        success: () => {
+          setIsUploaded(true);
+          return "Uploaded result to cloud";
+        },
+        error: (err) => {
+          return `Failed to upload result: ${err}`;
+        },
+      });
+    }
+  }
+
+  function isValidCloudExport(): boolean {
+    console.log(uploadInfo);
+    return true;
+  }
+
+  function runSequencer(sender: SendJsonMessage): void {
+    if (uploadAfterRun && !isValidCloudExport) {
+      toast.error("Please fill in the required fields to upload to cloud.");
+    }
+    setIsUploaded(false);
+    if (project === null) {
+      setIsLocked(true);
+      runRunnableSequencesFromCurrentOne(sender);
+    } else {
+      clearPreviousCycles();
+      // Run from the first runable sequence
+      const idx = sequences.findIndex((seq) => seq.runable);
+      if (idx !== -1) {
+        setIsLocked(true);
+        displaySequence(sequences[idx].project.name);
+        runRunnableSequencesFromCurrentOne(sender);
+      } else {
+        toast.info("No sequence selected to run.");
+      }
+    }
+  }
+
   const addNewElemsWithPermissions = withPermissionCheck(AddNewElems);
+  const setSequencesWithPermissions = withPermissionCheck(setSequences);
 
   return {
     elems,
     websocketId,
     setElems: setElemsWithPermissions,
     AddNewElems: addNewElemsWithPermissions,
     tree,
-    running,
-    markTestAsDone,
-    addTestToRunning,
     setIsLocked,
     setBackendState,
+    setBackendGlobalState,
     isLocked,
     isLoading,
     backendState,
+    backendGlobalState,
     setIsLoading,
     setUnsaved,
     isUnsaved,
     project,
-    setProject,
+    // Sequences
+    sequences,
+    setSequences: setSequencesWithPermissions,
+    runSequencer,
+    runNextRunnableSequence: runNextRunnableSequenceAndCycle,
+    displaySequence: displaySeq,
+    updateSequenceStatus,
+    addNewSequence: addNewSeq,
+    removeSequence,
+    // Cycles
+    cycleConfig,
+    cycleRuns,
+    setCycleCount,
+    setInfinite,
+    saveCycle,
+    diplayPreviousCycle,
+    displayNextCycle,
+    clearPreviousCycles,
+    // Upload
+    uploadAfterRun,
+    setUploadAfterRun,
+    uploadInfo,
+    handleUpload,
+    setStationId,
+    setSerialNumber,
+    setIntegrity,
+    setIsUploaded,
   };
```

Comment:
please DO NOT do this
each of the new functions you introduced can just be their own hook (or even just add it as a store action when possible! see `stores/project.ts`)
accessing store state should be as granular as possible

Reply:
I splitted this in two separated hooks, + removed all that were just "pass through". Those are now access with `sequencerStore(useShallow(state) => ...)`

Reply:
What left in the new SequencerState is:
- Run, Abort, Pause, Resume
- Upload
- AddNewSequence
- RemoveSequence (this one just so it's consistent with AddNewSequence being access through there)